driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/utils/DefaultConnectionConfigurator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Testerra
+ *
+ * (C) 2021, Mike Reiche,  T-Systems Multimedia Solutions GmbH, Deutsche Telekom AG
+ *
+ * Deutsche Telekom AG and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package eu.tsystems.mms.tic.testframework.utils;
+
+import java.net.URLConnection;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import javax.net.ssl.HttpsURLConnection;
+import org.openqa.selenium.WebDriver;
+
+public class DefaultConnectionConfigurator implements Consumer<URLConnection> {
```

Comment:
The `DefaultConnectionConfiguration` implements all the features of `FileDownloader` but abstract and reusable using `CertUtils` and `URLConnection`.